### PR TITLE
Allow extra nuttx config options via PX4_EXTRA_NUTTX_CONFIG env var

### PR DESCRIPTION
**Pull request by bkueng**
Our use-case is to enable nsh password protection for release builds.

For example:
```
export PX4_EXTRA_NUTTX_CONFIG="CONFIG_NSH_LOGIN_PASSWORD=\"test\";CONFIG_NSH_CONSOLE_LOGIN=y"
make px4_fmu-v5x
```

### Changelog Entry
For release notes:
```
Allow extra nuttx config options via PX4_EXTRA_NUTTX_CONFIG env var
```

### CODE REVIEW

File: platforms/nuttx/NuttX/CMakeLists.txt [hunk 1]
```diff
@@ -40,18 +40,30 @@ set(APPS_DIR ${NUTTX_SRC_DIR}/apps)
...
+	endforeach ()
+endif()
+
+
 # inflate .config
 add_custom_command(
 	OUTPUT ${NUTTX_DIR}/.config
 	COMMAND ${CMAKE_COMMAND} -E copy_if_different ${CMAKE_CURRENT_BINARY_DIR}/nuttx/Make.defs ${NUTTX_DIR}/Make.defs
-	COMMAND ${CMAKE_COMMAND} -E copy_if_different ${NUTTX_DEFCONFIG} ${NUTTX_DIR}/.config
+	COMMAND cat ${NUTTX_DEFCONFIG} ${EXTRA_NUTTX_CONFIG_FILE} > ${NUTTX_DIR}/.config
```
Comment (dagar) [inline platforms/nuttx/NuttX/CMakeLists.txt:58]: ${CMAKE_COMMAND} -E cat

File: platforms/nuttx/cmake/init.cmake [hunk 2]
```diff
@@ -77,6 +77,10 @@ execute_process(
 	OUTPUT_FILE ${CMAKE_CURRENT_BINARY_DIR}/nuttx_olddefconfig.log
 	RESULT_VARIABLE ret
 )
+execute_process(
+	COMMAND ${CMAKE_COMMAND} -E touch ${CMAKE_CURRENT_BINARY_DIR}/defconfig_inflate_stamp
+	WORKING_DIRECTORY ${NUTTX_DIR}
+)
```
Comment (dagar) [inline platforms/nuttx/cmake/init.cmake:83]: It probably doesn't matter for certain simple standalone cases, but I wonder if it should be done here as well for consistency.

Comment (bkueng) [reply platforms/nuttx/cmake/init.cmake:83]: I can add it if you prefer, it just gets a bit more complex.

File: platforms/nuttx/NuttX/CMakeLists.txt (hunk 1 above)
Comment (bkueng) [reply platforms/nuttx/NuttX/CMakeLists.txt:58]: That requires cmake 3.18, but Ubuntu 20.04 is still on 3.16

**Pull request merged**